### PR TITLE
NH-3817 - Test case and proposed fix

### PR DESCRIPTION
**Pull request by alobakov**
This contains both a test case to reproduce the issue and a fix to DefaultMergeListener.
Please refer to commit notes as well as https://nhibernate.jira.com/browse/NH-3817 for details.

### CODE REVIEW

File: src/NHibernate/Event/Default/DefaultMergeEventListener.cs [hunk 1]
```diff
@@ -71,8 +81,19 @@ public virtual void OnMerge(MergeEvent @event)
 							transientEntityName,
 							transientEntity.ToString());
 					}
+
+					// create a comma-delimited list of the entity names that failed to merge
```
Comment (hazzik) [inline src/NHibernate/Event/Default/DefaultMergeEventListener.cs]: Please use `string.Join(", ", transientEntityNames)`

Comment (alobakov) [inline src/NHibernate/Event/Default/DefaultMergeEventListener.cs]: Darn, it must have been so mind-boggling that I forgot the most obvious thing. :)
Thanks! - Will adjust that...

Comment (alobakov) [inline src/NHibernate/Event/Default/DefaultMergeEventListener.cs]: Hey Alex,
I'm not sure what to do next about this pull request. All of the platform-specific stuff that was failing the DB-specific tests has now been ironed-out, and yet some of the checks are still failing for unrelated reasons. Are there any further steps required on my part?

File: src/NHibernate.Test/NHSpecificTest/NH3817/Entity.cs [hunk 2]
```diff
@@ -0,0 +1,423 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NHibernate.Test.NHSpecificTest.NH3817
+{
+    public class Itinerary
+    {
+        #region Extensibility Method Definitions
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH3817/Entity.cs]: No #regions, please

**Comment by hazzik** [comment]
@alobakov can you please squash your pull request?

**Comment by alobakov** [comment]
@hazzik Sorry, do you mean close it? If closed, would it still be included in 4.1.0 regardless?

**Comment by hazzik** [comment]
I mean this: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
So, sqush the commits of pull request into one commit. Then force push.

**Comment by alobakov** [comment]
@hazzik Squashed and force-pushed. Sorry for being slow, - this is my very first use of Git.
BTW, not sure why all the checks are now failing, - at least 5 were completing successfully a couple days ago.

**Comment by hazzik** [comment]
>Squashed and force-pushed. Sorry for being slow, - this is my very first use of Git.

Perfectly done. Thanks.

>BTW, not sure why all the checks are now failing, - at least 5 were completing successfully a couple days ago.

Don't worry, it's TeamCity issue. It's just overloaded, and test timed out.

**Comment by alobakov** [comment]
@hazzik 
Hi Alex. Are any further steps required on my part (cleanup/etc.) to get this commit included in 4.1.0, or are we all set? - I'd be great to see it incorporated in an official release.
Thx

**Comment by oskarb** [comment]
I've fixed the indenting and merged. Unless there were any other issue, @hazzik ?

**Pull request merged**